src/gui/creds/webflowcredentials.cpp

```diff
@@ -227,6 +228,10 @@ bool WebFlowCredentials::stillValid(QNetworkReply *reply) {
     if (reply->error() != QNetworkReply::NoError) {
         qCWarning(lcWebFlowCredentials()) << reply->error();
         qCWarning(lcWebFlowCredentials()) << reply->errorString();
+        const auto accounts = AccountManager::instance()->accounts();
+	    for (auto account : accounts) {
+	        account->freshConnectionAttempt();
+	    }
```

Comment:
should you not call this only if the error is related to network issues ?
for example, imagine the desktop client doing a get request that will return a `404` result code, you do nto want to execute `freshConnectionAttempt()` after a `404`

an easy way to spot the issue is to have a client wihtout end-to-end encryption enabled but a server with the app installed and enabled

then if you open the settings dialog, we check if the server has a public end-to-end encryption key, gets a `404` and disconnects the session turning the tray icon to the disconnected state for some time before turning connected again

second thing is that most probably such a fix would be needed for all credentials classes not only when using the web flow v2 related code (i.e. this `WebFlowCredentials` class) but also other authentication methods